src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -229,6 +229,21 @@ private Bundle normalizeExtras(Context context, Bundle extras) {
                         Log.e(LOG_TAG, "normalizeExtras: JSON exception");
                     }
                 }
+            } else if (key.equals(APPCELERATOR_PAYLOAD)) {
+                try {
+                    String json = extras.getString(key);
```

Comment:
Hi, that is correct. Is amazon SNS related code is in the if key.equals("notification")?
Maybe I can reuse that part.